### PR TITLE
Add documentation for nixpkgs.config settings

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR is a follow-up PR for #5892 

It documents the following settings:
- `allowUnfree`
- `allowUnfreePredicate`
- `whitelistedLicenses`
- `blacklistedLicenses`

Feel free to suggest more documentation at the appropriate places.

### CODE REVIEW

**Comment by iElectric** [comment]
:beer: That's a good start. We have to document how it's used outside of NixOS. 

So given a file `~/.nixpkgs/config.nix`, you can declare:

```
{
  allowUnfree = true;
}
```

**Comment by geerds** [comment]
What are valid values for ```license```? 
```[ "gpl3" ]``` or ```with stdenv.lib.licenses; [ gpl3 ]``` ?

**Comment by matthiasbeyer** [comment]
@iElectric , @geerds I justed pushed fixups for the stuff you mentioned.

**Comment by iElectric** [comment]
Thanks, I'll take it from here :)

**Comment by matthiasbeyer** [comment]
I'd continue improving it, if you tell me what to improve!

Or do you mean you'll merge it?

**Comment by iElectric** [comment]
Thanks! Pushed and updated in 8408136da7bec07a17f8cf0760f9162df8041e4e

**Comment by matthiasbeyer** [comment]
Why not merging it via `Merge`?

File: doc/packageconfig.xml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+<chapter xmlns="http://docbook.org/ns/docbook"
+         xmlns:xlink="http://www.w3.org/1999/xlink"
+         xml:id="chap-packageconfig">
+
+<title>nixpkgs configuration</title>
```
Comment (edolstra) [inline doc/packageconfig.xml]: Title should be capitalized.

File: doc/packageconfig.xml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                <programlisting>
+                    nixpkgs.config.blacklistedLicenses = with stdenv.lib.licenses; [ agpl3 gpl3 ];
+                </programlisting>
+            </para>
+        </listitem>
+    </itemizedlist>
+
+    <para>
+        A complete list of licenses can be found in the file
+        <programlisting>
```
Comment (edolstra) [inline doc/packageconfig.xml]: This should be `<filename>` rather than `<programlisting>`.

File: doc/packageconfig.xml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        The Nix package manager can be configured to allow or deny certain
+        package sets. At this moment, packages can either be allowed to be
+        installed or denied to be installed based on their license.
+    </para>
+
+    <itemizedlist>
+        <listitem>
+            <para>
+                Allow packages that do not have a free license by setting
+                <programlisting>
```
Comment (edolstra) [inline doc/packageconfig.xml]: `<programlisting>` should not be intended, because whitespace is significant.

File: doc/packageconfig.xml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            </para>
+        </listitem>
+    </itemizedlist>
+
+    <para>
+        A complete list of licenses can be found in the file
+        <programlisting>
+            lib/licenses.nix
+        </programlisting>
+        of the nix package tree.
```
Comment (edolstra) [inline doc/packageconfig.xml]: `nix package tree` -> `Nixpkgs source tree`.

File: doc/packageconfig.xml (hunk 2 above)
Comment (iElectric) [inline doc/packageconfig.xml]: See https://github.com/NixOS/nixpkgs/commit/8408136da7bec07a17f8cf0760f9162df8041e4e

**Pull request closed without merging**